qiskit_experiments/library/characterization/T2Hahn.py

```diff
@@ -0,0 +1,188 @@
+
+         Raises:
+             Error for invalid input.
+        """
+        # Initialize base experiment
+        super().__init__([qubit])
+        # Set configurable options
+        self.set_experiment_options(
+            delays=delays, n_echos=n_echos, phase_alt_echo=phase_alt_echo, qubit=qubit
+        )
+        self._verify_parameters()
+
+    def _verify_parameters(self):
+        """
+        Verify input correctness, raise QiskitError if needed.
+        Args:
+            qubit: the qubit under test.
+
+        Raises:
+            QiskitError : Error for invalid input.
+        """
+        if any(delay <= 0 for delay in self.experiment_options.delays):
+            raise QiskitError(
+                f"The lengths list {self.experiment_options.delays} should only contain "
+                "positive elements."
+            )
+        if len(set(self.experiment_options.delays)) != len(self.experiment_options.delays):
+            raise QiskitError(
+                f"The lengths list {self.experiment_options.delays} should not contain "
+                "duplicate elements."
+            )
+
+        if any(
+            self.experiment_options.delays[idx - 1] >= self.experiment_options.delays[idx]
+            for idx in range(1, len(self.experiment_options.delays))
+        ):
+            raise QiskitError(
+                f"The number of identity gates {self.experiment_options.delays} should "
+                "be increasing."
+            )
+
+        if isinstance(self.experiment_options.qubit, list):
+            if len(self.experiment_options.qubit) != 1:
+                raise QiskitError(
+                    "The experiment if for 1 qubit. For multiple qubits,"
+                    " please use parallel experiments."
+                )
+            if self.experiment_options.qubit[0] < 0:
+                raise QiskitError(
+                    f"The index of the qubit {self.experiment_options.qubit[0]} should "
+                    "be non-negative."
+                )
+        else:
+            if self.experiment_options.qubit < 0:
+                raise QiskitError(
+                    f"The index of the qubit {self.experiment_options.qubit} should "
+                    "be non-negative."
+                )
+
+        if self.experiment_options.n_echos < 1:
+            raise QiskitError(
+                f"The number of echoes {self.experiment_options.n_echos} should " "be at least 1."
+            )
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """
+        Args:
+            backend: Optional, a backend object.
+
+        Returns:
+            The experiment circuits.
+
+        """
+
+        circuits = []
+        qubit = list(self._physical_qubits)
+        for circ_index, delay in enumerate(self.experiment_options.delays):
+            circ = QuantumCircuit(max(qubit) + 1, len(qubit))
+            circ.name = "t2circuit_" + str(circ_index) + "_0"
```

Comment:
This code has been copied from Ignis. Adjust it to Experiments, look at other similar experiments in qiskit-experiments.

Reply:
Made it to look like t2ramsey.